main/integration-tests/__test__/api-tests/workspace-types/configurations/get-configurations.test.js

```diff
@@ -0,0 +1,151 @@
+  describe('Get configurations', () => {
+    it('should fail to get all configurations if user is inactive', async () => {
+      const adminSession2 = await setup.createAdminSession();
+      const workspaceTypeId = setup.gen.string({ prefix: 'workspace-test' });
+
+      await adminSession.resources.workspaceTypes.create({
+        id: workspaceTypeId,
+      });
+
+      await adminSession.resources.users.deactivateUser(adminSession2.user);
+
+      await expect(
+        adminSession2.resources.workspaceTypes
+          .workspaceType(workspaceTypeId)
+          .configurations()
+          .getAll(),
+      ).rejects.toMatchObject({
+        code: errorCode.http.code.unauthorized,
+      });
+    });
+  });
+
+  it('should fail to get all configurations if user is anonymous', async () => {
+    const anonymousSession = await setup.createAnonymousSession();
+    const workspaceTypeId = setup.gen.string({ prefix: 'workspace-test' });
+
+    await adminSession.resources.workspaceTypes.create({
+      id: workspaceTypeId,
+    });
+
+    await expect(
+      anonymousSession.resources.workspaceTypes
+        .workspaceType(workspaceTypeId)
+        .configurations()
+        .getAll(),
+    ).rejects.toMatchObject({
+      code: errorCode.http.code.badImplementation,
+    });
+  });
+
+  it('should return all configurations if user is admin', async () => {
+    const workspaceTypeId = setup.gen.string({ prefix: 'workspace-test' });
+
+    await adminSession.resources.workspaceTypes.create({
+      id: workspaceTypeId,
+    });
+
+    const configurationId = setup.gen.string({ prefix: 'configuration-test' });
+
+    await adminSession.resources.workspaceTypes
+      .workspaceType(workspaceTypeId)
+      .configurations()
+      .create({
+        id: configurationId,
+      });
+
+    await expect(
+      adminSession.resources.workspaceTypes
+        .workspaceType(workspaceTypeId)
+        .configurations()
+        .getAll(),
+    ).resolves.toEqual(expect.arrayContaining([expect.objectContaining({ id: configurationId })]));
+  });
+
+  it('should not return configurations if a user does not have access', async () => {
+    const workspaceTypeId = setup.gen.string({ prefix: 'workspace-test' });
+
+    await adminSession.resources.workspaceTypes.create({
+      id: workspaceTypeId,
+      status: 'approved',
+    });
+
+    const configurationId = setup.gen.string({ prefix: 'configuration-test' });
+
+    await adminSession.resources.workspaceTypes
+      .workspaceType(workspaceTypeId)
+      .configurations()
+      .create({
+        id: configurationId,
+        denyRoleIds: ['researcher'],
+      });
+
+    const researcherSession = await setup.createResearcherSession();
+
+    await expect(
+      researcherSession.resources.workspaceTypes
+        .workspaceType(workspaceTypeId)
+        .configurations()
+        .getAll(),
+    ).resolves.toHaveLength(0);
+  });
+
+  it('should return configurations a user has access', async () => {
```

Comment:
Minor, did you mean 'should return configurations when user has access'.  Can be fixed in the next PR.

Reply:
Fixed. Thanks for catching this!